.github/workflows/test.yml

```diff
@@ -0,0 +1,86 @@
+name: Run tests
+
+on:
+  push:
+    branches: [main]
+
+  pull_request:
+    branches: [main]
+
+jobs:
+  run-npm-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [19.x]
+        python-version: ["3.10"]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "npm"
+
+      - name: Install Node dependencies
+        run: |
+          npm install
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install . --no-deps
+          python -m pip install -r requirements.txt
+          python -m pip install -r requirements-dev.txt
+
+      - name: Run pre-commit
+        run: |
+          pre-commit run --all-files --verbose --show-diff-on-failure
+
+      - name: Run Node tests
+        run: |
+          npm ci
+          npm run jsdeps
+          npm run build --if-present
+          npm test
+
+  run-py-test:
```

Comment:
Please add a "needs" dependency on the run-npm-test job:
- https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds

This can (should?) be done before the runs-on info.

This ensures that we only run pytest after the npm tests succeed. It's mostly important for the pre-commit step: we don't want to start pytest before knowing pre-commit was successful.